[hooks] Document user-defines in pubspec (#7323)
dart-lang/site-www: src/content/tools/pub/pubspec.md

For more information, check out
[Security advisories](/tools/pub/security-advisories).

### Hooks

To configure package-specific hooks, such as native build or link hooks,
use the `hooks` field.
Under the `hooks` field,

> you can configure parameters that customize how hooks run.

Reviewer: ```suggestion
Under the `hooks` field,
configure parameters that customize how hooks run.
```

For example:

```yaml
hooks:
  user_defines:
    my_package:
      enable_experimental: true
```

> For more information, check out the [Hooks](/tools/hooks) page.

Reviewer 2: Should this direct them to the more directly related content?

```suggestion
To learn more about configuring hooks,
check out [Hook configuration][].

[Hook configuration]: /tools/hooks#hook-configuration
```
